docs/en/latest/plugins/opa.md

```diff
@@ -0,0 +1,275 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## Summary
+
+- [**Description**](#description)
+- [**Attributes**](#attributes)
+- [**Data Definition**](#data-definition)
+- [**Example**](#example)
+
+## Description
+
+The `opa` plugin is used to integrate with Open Policy Agent. By using this plugin, users can decouple functions such as authentication and access to services and reduce the complexity of the application system.
+
+## Attributes
+
+| Name | Type | Requirement | Default | Valid | Description |
+| -- | -- | -- | -- | -- | -- |
+| host | string | required |   |   | Open Policy Agent service host (eg. https://localhost:8181) |
+| ssl_verify | boolean | optional | true |   | Whether to verify the certificate |
+| policy | string | required |   |   | OPA policy path (It is a combination of `package` and `decision`. When you need to use advanced features such as custom response, `decision` can be omitted) |
+| timeout | integer | optional | 60000ms | [1, 60000]ms | OpenWhisk Action and HTTP call timeout. |
+| keepalive | boolean | optional | true |   | HTTP keepalive |
+| keepalive_timeout | integer | optional | 60000ms | [1000,...] | keepalive idle timeout |
+| keepalive_pool | integer | optional | 5 | [1,...] | Connection pool limit |
+| with_route | boolean | optional | false |   | Whether to send information about the current route. |
+| with_service | boolean | optional | false |   | Whether to send information about the current service. |
+| with_consumer | boolean | optional | false |   | Whether to send information about the current consumer. (It may contain sensitive information such as apikey, so please turn it on again if you are sure it is safe) |
+
+## Data Definition
+
+### APISIX to OPA service
+
+The `type` indicates that the request type. (e.g. `http` or `stream`)
+The `reqesut` is used when the request type is `http`, it contains the basic information of the request. (e.g. url, header)
+The `var` contains basic information about this requested connection. (e.g. IP, port, request timestamp)
+The `route`, `service`, and `consumer` will be sent only after the `opa` plugin has enabled the relevant features, and their contents are same as those stored by APISIX in etcd.
+
+```json
+{
+    "type": "http",
+    "request": {
+        "scheme": "http",
+        "path": "\/get",
+        "headers": {
+            "user-agent": "curl\/7.68.0",
+            "accept": "*\/*",
+            "host": "127.0.0.1:9080"
+        },
+        "query": {},
+        "port": 9080,
+        "method": "GET",
+        "host": "127.0.0.1"
+    },
+    "var": {
+        "timestamp": 1700000000,
```

Comment:
Better to use a format like 170123... so people don't need to count the zero.

Reply:
changed